nixos/modules/tasks/network-namespaces.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+with lib; {
+  options = {
+    networking.namespaces = mkOption {
+      default = [];
+      type = types.listOf types.str;
+      description = ''
+        A list of named network namespaces to create.
+      '';
+    };
+  };
+  config.systemd.services = builtins.listToAttrs (map (netns: {
+    name = "network-namespace-${netns}";
+    value = {
+      description = "Network Namespace: ${netns}";
+      before = [ "network-pre.target" ];
+      wantedBy = [ "multi-user.target" ];
```

Comment:
Nit: According to [this documentation page](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/), one should set `Wants=network-pre.target`.

I don't t know if there will be a difference in practice for most users, but I do think there's some dissonance in binding namespace creation to the `multi-user.target`.

OTOH the wireguard module also does `WantedBy=multi-user.target`, so maybe this is a convention that makes sense in NixOS?

Reply:
Citing from above docs:

> Services that want to be run before the network is configured should place Before=network-pre.target and also set Wants=network-pre.target to pull it in.

We should add `Wants=network-pre.target`.

I don't understand the question regarding `multi-user.target`.